### ISSUE TITLE
Spyder freezes, stops running

### ISSUE THREAD

**Issue by Helge99** [spyder-ide/spyder#15821]
### What steps will reproduce the problem?

nothing specific that will cause Spyder to freeze. Although I have encountered it since running permutations from itertools, this last time I only ran dir(scipy)

## Versions

* Spyder version: 4.2.5 
* Python version: 3.8.3
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0            :  1.4.0 (OK)
chardet >=2.0.0                 :  4.0.0 (OK)
cloudpickle >=0.5.0             :  1.6.0 (OK)
diff_match_patch >=20181111     :  20200713 (OK)
intervaltree >=3.0.2            :  3.1.0 (OK)
IPython >=7.6.0                 :  7.21.0 (OK)
jedi =0.17.2                    :  0.17.2 (OK)
jsonschema >=3.2.0              :  3.2.0 (OK)
keyring >=17.0.0                :  22.3.0 (OK)
nbconvert >=4.0                 :  6.0.7 (OK)
numpydoc >=0.6.0                :  1.1.0 (OK)
paramiko >=2.4.0                :  2.7.2 (OK)
parso =0.7.0                    :  0.7.0 (OK)
pexpect >=4.4.0                 :  4.8.0 (OK)
pickleshare >=0.4               :  0.7.5 (OK)
psutil >=5.3                    :  5.8.0 (OK)
pygments >=2.0                  :  2.8.1 (OK)
pylint >=1.0                    :  2.7.2 (OK)
pyls >=0.36.2;<1.0.0            :  0.36.2 (OK)
pyls_black >=0.4.6              :  0.4.6 (OK)
pyls_spyder >=0.3.2             :  0.3.2 (OK)
qdarkstyle >=2.8;<3.0           :  2.8.1 (OK)
qtawesome >=0.5.7               :  1.0.2 (OK)
qtconsole >=5.0.3               :  5.0.3 (OK)
qtpy >=1.5.0                    :  1.9.0 (OK)
rtree >=0.8.3                   :  0.9.4 (OK)
setuptools >=39.0.0             :  52.0.0.post20210125 (OK)
sphinx >=0.6.6                  :  3.5.3 (OK)
spyder_kernels >=1.10.2;<1.11.0 :  1.10.2 (OK)
textdistance >=4.2.0            :  4.2.1 (OK)
three_merge >=0.1.1             :  0.1.1 (OK)
watchdog >=0.10.3;<2.0.0        :  1.0.2 (OK)
zmq >=17                        :  20.0.0 (OK)

# Optional:
cython >=0.21                   :  0.29.22 (OK)
matplotlib >=2.0.0              :  3.3.1 (OK)
numpy >=1.7                     :  1.19.2 (OK)
pandas >=1.1.1                  :  1.2.0 (OK)
scipy >=0.17.0                  :  1.6.2 (OK)
sympy >=0.7.3                   :  1.7.1 (OK)
```

**Comment by dalthviz** [comment]
Hi @Helge99 thanks for the feedback! That's strange, is this something happening to you consistently? and is there any other condition for the freeze to happen (are you running something else that could be consuming machine resources)?

Let us know :)

**Comment by Helge99** [comment]
Hi Daniel,
yes, it is strange and I haven't had any problems until now with Spyder.
And I only encountered freezes after I downloaded and ran
itertools.permutations, every time I ran it Spyder froze.

Since then I've uninstalled and reinstalled Spyder and it seems to be
running without any issues now.

I checked the CPU and memory consumption, both were fine until I ran
itertools.permutations, and memory suddenly hit close to or even 100%. I
have an XPS17 with an i7 CPU and 16 GB internal memory.

I hope it won't happen again, but I will notify you if it does.

Thanks,
Mikael

Den ons 9 juni 2021 kl 18:41 skrev Daniel Althviz Moré <
***@***.***>:
